packages/compiler/lib/decorators.ts

```diff
@@ -961,7 +961,7 @@ export function getOverloadedOperation(
   return program.stateMap(overloadsOperationKey).get(operation);
 }
 
-const projectedNameKey = Symbol("projectedNameKey");
+const projectedNameKey = createStateSymbol("projectedNameKey");
```

Comment:
This one is not related to the symptom in #1625, but could result in the same sort of bug.